### ISSUE TITLE
SourceForge preferred URL formats not working for MPlayerX 1.1.3

### ISSUE THREAD

**Issue by grootknol** [caskroom/homebrew-cask#21911]
There is a new version of MPlayerX out that can be downloaded through the URL [http://liquidtelecom.dl.sourceforge.net/project/mplayerx-osx/MPlayerX-1.1.3-1913.zip](http://liquidtelecom.dl.sourceforge.net/project/mplayerx-osx/MPlayerX-1.1.3-1913.zip), but cask-repair won't accept it because the URL is not in a SourceForge preferred format.

I've tried the preferred formats with MplayerX and they don't work, nor did the one in the current cask using only a new version number ("1.1.0" --> "1.1.3-1913").

What should I do?

**Comment by vitorgalvao** [comment]
It’s not exactly `cask-repair` that isn’t allowing it, but `brew cask audit` (`cask-repair` just refuses to submit if `audit` does not pass).

Those required formats were introduced a long while ago, when people were frequently submitting specific mirrors, instead of the more generic URLs that pick the best mirror.

I’m looking into it.

**Comment by vitorgalvao** [comment]
Where are you getting that URL from? The website links to `https://download-codeplex.sec.s-msft.com/Download/Release?ProjectName=mpx&DownloadId=1566333&FileTime=131099519222070000&Build=21031` for the download.

**Comment by grootknol** [comment]
The author announced last year that he would start to use the installer to monetize the app (adware), but the version at SourceForge is clean, so I download from there, as does the MplayerX.

**Comment by vitorgalvao** [comment]
> start to use the installer to monetize the app (adware)

Holy crap, MplayerX really is turning to shit. I remember when it was the best we had.

Have you tried [mpv](https://mpv.io/)? I highly recommend it as the new best. Install with `brew install mpv --with-bundle`. They don’t release OS X bundles all that often themselves, which is why we don’t have a cask.

> but the version at SourceForge is clean

[How ironic](http://arstechnica.com/information-technology/2015/05/sourceforge-grabs-gimp-for-windows-account-wraps-installer-in-bundle-pushing-adware/).

Still, is there somewhere on the official site we can confirm the sourceforge links as being official?

**Comment by grootknol** [comment]
Yeah, I knew that Sourceforge has been hijacking abandoned software and
bundling them in installers with adware, but MPlayerX is distributed as a
ZIP there so I think the Sourceforge version is ok.

I don't know how I found about it before (I didn't use brew in my previous
OS X setup), but looking around I've found:

   -
   http://www.thesafemac.com/mplayerx-adware-behaving-like-malware/#comment-38674
   - http://www.macupdate.com/app/mac/33261/mplayerx

Besides, the current cask downloads from Sourceforge too, even if the URL
is different than the one I provided.
Mine is the URL that Sourceforge directs the browser to in order to
download the file. I didn't try if the previous scheme still works for
1.1.0, but it didn't work with 1.1.3, so I went after what Sourceforge
would provide if I was manually downloading the file.



*--Ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn*
